From: Ahmad Fatoum <a.fatoum@pengutronix.de>
Date: Mon, 21 Mar 2022 16:48:49 +0100
Subject: [PATCH] lxatac: make DT extension in bootloader easier

barebox will need to extend some nodes, so add labels and aliases to
make it easier to reference them. The alternative is to reference by
path, which is unnecessarily verbose and more likely to break.

While at it, add a further more descriptive label for the SiP EEPROM,
as this will be the EEPROM used for Baseboard factory data.

Signed-off-by: Ahmad Fatoum <a.fatoum@pengutronix.de>
---
 arch/arm/boot/dts/stm32mp157c-lxa-tac.dts | 20 +++++++++++++-------
 1 file changed, 13 insertions(+), 7 deletions(-)

diff --git a/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts b/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts
index 2d48bf42a6b9..03c8aa42302b 100644
--- a/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts
+++ b/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts
@@ -21,6 +21,9 @@ / {
 
 	aliases {
 		ethernet0 = &ethernet0;
+		/* Enable MAC address fixup in absence of bootloader DSA driver */
+		ethernet1 = &port_uplink;
+		ethernet2 = &port_dut;
 		mmc1 = &sdmmc2;
 		serial0 = &uart4;
 		serial1 = &usart3;
@@ -45,7 +48,7 @@ led-1 {
 			gpios = <&gpiog 7 GPIO_ACTIVE_HIGH>;
 		};
 
-		led-2 {
+		led_status_dut: led-2 {
 			label = "tac:green:statusdut";
 			gpios = <&gpioa 13 GPIO_ACTIVE_LOW>;
 			linux,default-trigger = "default-on";
@@ -407,7 +410,7 @@ &i2c1 {
 	pinctrl-1 = <&i2c1_sleep_pins_b>;
 	status = "okay";
 
-	pwr_gpio: pca9570@48 {
+	powerboard_gpio: pca9570@48 {
 		compatible = "nxp,pca9570";
 		reg = <0x24>;
 		gpio-controller;
@@ -416,7 +419,7 @@ pwr_gpio: pca9570@48 {
 		status = "okay";
 	};
 
-	pwr_eeprom@50 {
+	powerboard_eeprom: eeprom@50 {
 		compatible = "atmel,24c02";
 		reg = <0x50>;
 		vcc-supply = <&v3v3>;
@@ -429,6 +432,9 @@ lm75ad@48 {
 	};
 };
 
+baseboard_eeprom: &sip_eeprom {
+};
+
 &i2c5 {
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&i2c5_pins_b>;
@@ -454,7 +460,7 @@ lmp92064@0 {
 		compatible = "ti,lmp92064";
 		reg = <0>;
 		spi-max-frequency = <5000000>;
-		reset-gpios = <&pwr_gpio 2 GPIO_ACTIVE_HIGH>;
+		reset-gpios = <&powerboard_gpio 2 GPIO_ACTIVE_HIGH>;
 		shunt-resistor = <15000>;
 	};
 };
@@ -482,15 +488,15 @@ ksz9563: ksz9563@0 {
 		ports {
 			#address-cells = <1>;
 			#size-cells = <0>;
-			port@0 {
+			port_dut: port@0 {
 				reg = <0>;
 				label = "dut";
 			};
-			port@1 {
+			port_uplink: port@1 {
 				reg = <1>;
 				label = "uplink";
 			};
-			port@2 {
+			port_cpu: port@2 {
 				reg = <2>;
 				label = "cpu";
 				ethernet = <&ethernet0>;
